### PR TITLE
Allow users with read permissions to view Integrations tabs

### PR DESCRIPTION
**Pull request by CodyWMitchell**
### Description

Update the `/settings/integrations` page so that the Communications, Reporting and Automation, and Webhooks tabs are visible for users with `integrations:endpoints:read` or `integrations:*:*` permissions instead of only Org Admins.

[RHCLOUD-30431](https://issues.redhat.com/browse/RHCLOUD-30431)

(Merge only after https://github.com/RedHatInsights/notifications-frontend/pull/543 is merged)

---

### Screenshots

image

---

### Checklist ☑️
- [x] PR only fixes one issue or story 
- [x] Change reviewed for extraneous code 
- [x] UI best practices adhered to 
- [x] Commits squashed and meaningfully named 
- [x] All PR checks pass locally (build, lint, test, E2E)

##
- [ ] _(Optional) QE: Needs QE attention (OUIA changed, perceived impact to tests, no test coverage)_
- [ ] _(Optional) QE: Has been mentioned_
- [ ] _(Optional) UX: Needs UX attention (end user UX modified, missing designs)_
- [ ] _(Optional) UX: Has been mentioned_
##

### CODE REVIEW

File: src/redux/user/actions.js [hunk 1]
```diff
@@ -26,6 +26,33 @@ export const loadWritePermissions = (getUserPermissions) => (dispatch) => {
     );
 };
 
+export const loadReadIntegrationsEndpointsPermissions = (getUserPermissions) => (dispatch) => {
+  dispatch({ type: ACTION_TYPES.SET_READ_INTEGRATIONS_ENDPOINTS_PERMISSIONS_PENDING });
+
+  return getUserPermissions('integrations', true) // bypassCache = true
+    .then((permissions) => {
+      const allPermission = permissions.reduce((acc, curr) => [...acc, curr?.permission], []);
```
Comment (karelhala) [inline src/redux/user/actions.js:34]: No need to use reduce and copy object.
```suggestion
      const allPermission = permissions.map((curr) => curr?.permission);
```

File: src/redux/user/actions.js [hunk 2]
```diff
@@ -26,6 +26,33 @@ export const loadWritePermissions = (getUserPermissions) => (dispatch) => {
     );
 };
 
+export const loadReadIntegrationsEndpointsPermissions = (getUserPermissions) => (dispatch) => {
+  dispatch({ type: ACTION_TYPES.SET_READ_INTEGRATIONS_ENDPOINTS_PERMISSIONS_PENDING });
+
+  return getUserPermissions('integrations', true) // bypassCache = true
```
Comment (karelhala) [inline src/redux/user/actions.js:32]: What you can do is actually flip the order of actions and dispatching a promise.
```JS
dispatch({
  type: ACTION_TYPES.SET_READ_INTEGRATIONS_ENDPOINTS_PERMISSIONS,
  payload: getUserPermissions('integrations', true).then((permissions) => {
    const allPermission = permissions.map((curr) => curr?.permission);
    return allPermission.includes('integrations:*:*') || allPermission.includes('integrations:endpoints:read');
  })
})
```

**Review by karelhala — COMMENTED** [review]
Looking good! Just a slight change in dispatching action

File: src/redux/user/actions.js (hunk 2 above)
Comment (CodyWMitchell) [reply src/redux/user/actions.js]: really cool! I didn't know you could do this

**Review by karelhala — APPROVED** [review]

**Comment by karelhala** [comment]
Looks like there's `ERR_UNHANDLED_REJECTION` in your code somwhere.

**Review by karelhala — APPROVED** [review]

**Pull request merged**